### PR TITLE
doc: docs for cardinality benchmarking

### PR DESCRIPTION
**Pull request by patrickhyw**
**Summary**: Updated `README.md`, `SUMMARY.md`, and a new file `cost_model_benchmarking.md` to document information about cardinality benchmarking.

**Details**:
* `README.md` contains a quickstart command.
* `cost_model_benchmarking.md` contains conceptual info and notes about operating and extending the system.
* I name it "benchmarking" instead of "testing" in the docs to distinguish it from functional testing. I renamed `perftest` and `cardtest` to `perfbench` and `cardbench` to match how we're calling it "benchmarking" instead of "testing".

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +25,13 @@ You can also run the Datafusion cli to interactively experiment with optd.
 cargo run --bin datafusion-optd-cli
 ```
 
+You can also test the performance of the cost model with the "cardinality testing" feature (more info in the [docs](docs/)).
```
Comment (AlSchlo) [inline README.md]: Not consistent: cardinality benchmarking right?

File: docs/src/cost_model_benchmarking.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Cost Model Cardinality Benchmarking
+
+## Overview
+You can benchmark the cardinality estimates of optd's cost model against other DBMSs using the optd-perfbench module.
+
+All aspects of benchmarking except for setting up comparison DBMSs–loading workload data, building statistics, gathering the true cardinality of workload queries, running explains on workload queries, and aggregating results–are handled automatically.
```
Comment (AlSchlo) [inline docs/src/cost_model_benchmarking.md]: Confusing statement. It looks like we do nothing? (I mean the except part).

File: docs/src/cost_model_benchmarking.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+We elected not to automate the installation and setup of the DBMS in order to accomodate the needs of all users. For instance, some users prefer installing Postgres on Homebrew, others choose to install the Mac application, while others wish to create a Postgres Docker container.
```
Comment (AlSchlo) [inline docs/src/cost_model_benchmarking.md]: I still think having a docker dependency is alright :-)

**Review by AlSchlo — APPROVED** [review]
Thanks for writing this!

File: README.md (hunk 1 above)
Comment (patrickhyw) [reply README.md]: good catch!

File: docs/src/cost_model_benchmarking.md (hunk 2 above)
Comment (patrickhyw) [reply docs/src/cost_model_benchmarking.md]: thanks, fixed

File: docs/src/cost_model_benchmarking.md (hunk 3 above)
Comment (patrickhyw) [reply docs/src/cost_model_benchmarking.md]: good point. I added a note for this for the future

**Pull request merged**